lib/ble.toit

```diff
@@ -237,12 +246,12 @@ class Client:
   resource_state_/monitor.ResourceState_
 
   constructor .device .address:
-    ble_connect_ device.resource_group_ address.raw_
-    state := device.resource_state_.wait_for_state CONNECTED_EVENT_ | CONNECTED_EVENT_
-    if state & CONNECT_FAILED_EVENT_ != 0:
-      throw "BLE connection failed"
     gatt_ = ble_get_gatt_ device.resource_group_
     resource_state_ = monitor.ResourceState_ device.resource_group_ gatt_
+    ble_connect_ device.resource_group_ address.raw_ gatt_
+    state := resource_state_.wait_for_state CONNECTED_EVENT_ | CONNECT_FAILED_EVENT_
+    if state & CONNECT_FAILED_EVENT_ != 0:
+      throw "BLE connection failed"
```

Comment:
Should we dispose the resource state and close the resource group here to clean up nicely?

Reply:
Yes. Please advice how that should be done.